enumerables.rb

```diff
@@ -0,0 +1,66 @@
+# enumerable.rb
+module Enumerable
+  def my_each
+    i = 0
+    while to_a.length > i
+      yield(to_a[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while to_a.length > i
+      yield(to_a[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    to_a.my_each { |item| new_array.push(item) if yield(item) }
+    new_array
+  end
+
+  def my_all?
+    to_a.my_each { |item| return false if yield(item) == false }
+    true
+  end
+
+  def my_any?
+    to_a.my_each { |item| return true if yield(item) }
+    false
+  end
+
+  def my_none?
+    to_a.my_each { |item| return false if yield(item) }
+    true
+  end
```

Comment:
- [ ]  enumerables #my_none? when no block or argument is given returns true only if none of the collection members is true
      Failure/Error: expect(false_array.my_none?).to be true

- [ ] enumerables #my_none? when no block or argument is given returns false only if one of the collection members is true
      Failure/Error: expect(true_array.my_none?).to be false
- [ ] enumerables #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
      Failure/Error: expect(array.my_none?(String)).to be array.my_none?(String)

- [ ]  enumerables #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric
      Failure/Error: expect(array.my_none?(Numeric)).to be array.my_none?(Numeric)

- [ ] enumerables #my_none? when a class is passed as an argument returns false if any of the collection is a member of such class
      Failure/Error: expect(array.my_none?(String)).to be array.my_none?(String)

- [ ] enumerables #my_none? when a Regex is passed as an argument returns true if none of the collection matches the Regex
      Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)

- [ ]  enumerables #my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex
      Failure/Error: expect(words.my_none?(/d/)).to be words.none?(/d/)

- [ ] enumerables #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)

- [ ]  enumerables #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)

Reply:
Everything is done!!! Sorry for not checking all the boxes but I am unable to check all of them on GitHub, don't know why though